libs/trainer/bbhnet/trainer/trainer.py

```diff
@@ -271,4 +283,23 @@ def train(
                     )
                     break
 
+        # if no validation dataset passed, save model based on
+        # best training loss. i.e. attempt to overfit
+        else:
```

Comment:
Not entirely sure this is necessary: for any sufficiently complex model, your train loss will almost always be able to approach 0 arbitrarily. But it's probably fine to keep in here for now

Reply:
Yeah true actually this is probably important for LR scheduling since we're not doing it based on epoch: as you approach a minimum you'll want to take smaller steps to avoid stepping right over it and sort of "circling the drain". Normally if you use like a cosine or exponential decay this isn't an issue since the LR decreases on its own, but since we use a plateau LR scheduler you'll still need to manage this manually.

I guess more broadly this is probably less important just because it will generally be the case that if you're not doing validation, this probably isn't a run you're doing to achieve good results but to sanity check your training code, and in that sense this probably doesn't matter too much. But either way it's definitely fine to include for now since it doesn't negatively impact anything